### PR TITLE
Fixed the validations and flash messages.

### PR DESCRIPTION
**Pull request by icorson3**
- Removed excess CSS that was giving me an annoying bar at the top of my page.
- Fixed the validations for my games and the ability to add a game. 
- Error messages will now show if the wrong information is entered.

### CODE REVIEW

**Comment by icorson3** [comment]
I am merging so that I can continue to work! But please let me know if you have suggestions.

File: app/controllers/api/v1/games_controller.rb [hunk 1]
```diff
@@ -9,9 +9,13 @@ def show
...
   def create
-    game = Game.create(game_params)
-    User.assign_game(current_user.id, game.id)
-    render json: game
+    @game = Game.new(game_params)
+    if @game.save
+      User.assign_game(current_user.id, @game.id)
+      render json: @game
+    else
+      render json: { errors: @game.errors }, status: 500
```
Comment (dalexj) [inline app/controllers/api/v1/games_controller.rb:17]: I think the status should be a 400-level, not 500. probably `:unprocessable_entity`

File: app/views/users/_new_game_form.html.erb [hunk 2]
```diff
@@ -6,10 +6,11 @@
   <div class="form-group" id="new-game">
     <input class="form-control hidden" type="text" id="current-user" value="<%= current_user.id %>">
     <input class="form-control hidden" type="text" id="current-role" value="<%= current_user.role %>">
-    <input class="form-control" type="text" id="game-name" value='Name'>
-    <input class="form-control" type="url" id="game-heroku" value='Game Link: Must be the full url'>
-    <input class="form-control" type="url" id="game-github" value = 'Github URL: Must be the full url'>
-    <input class="form-control" type="url" id="game-screenshot" value= "Gif: Must be the full url">
+    <input class="form-control" type="text" id="game-name" placeholder='Name'>
+    <input class="form-control" type="url" id="game-heroku" placeholder='Game Link: Must be the full url'>
+    <input class="form-control" type="url" id="game-github" placeholder = 'Github URL: Must be the full url'>
```
Comment (dalexj) [inline app/views/users/_new_game_form.html.erb:11]: some html styling: use double quotes, and keep equals signs without spaces on either side

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
+    <h3><%= "#{@user.username}'s Games" %></h3>
+      <%= render partial: 'new_game_form' %>
   </div>
-</div>
+<% else %>
+  <div  id="guest-login">
+    <p> The Turing Arcade was created as a portal for games created by <%= link_to 'Turing', 'https://www.turing.io/', target: 'blank' %> Students.</br></br>
+      To see the work our students have done:</br></br>
+    <%= link_to "Browse All Games", games_path, style: "color: black", class: "browse_css_class"%></br></br>
+    To add your own game, you must sign in through <%= link_to 'github', github_login_path %>.</p>
```
Comment (dalexj) [inline app/views/users/show.html.erb:12]: try to keep your ruby code consistent, theres still debate of double vs single quotes, but you should pick 1 and stick with it. I find it very useful to tell the difference between ruby/javascript with using double quote for ruby and single for JS

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-    <% end %>
+<% if @user %>
+  <div class="user-show" id="welcome">
+    <h2>Welcome, <%= @user.username %></h2>
+    <h3><%= "#{@user.username}'s Games" %></h3>
+      <%= render partial: 'new_game_form' %>
   </div>
-</div>
+<% else %>
+  <div  id="guest-login">
```
Comment (dalexj) [inline app/views/users/show.html.erb:8]: annoying nitpick but 2 spaces

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
-        <p> The Turing Arcade was created as a portal for games created by <%= link_to 'Turing', 'https://www.turing.io/', target: 'blank' %> Students.</br></br>
-          To see the work our students have done:</br></br>
-        <%= link_to "Browse All Games", games_path, style: "color: black", class: "browse_css_class"%></br></br>
-        To add your own game, you must sign in through <%= link_to 'github', github_login_path %>.</p>
-      </div>
-    <% end %>
+<% if @user %>
+  <div class="user-show" id="welcome">
+    <h2>Welcome, <%= @user.username %></h2>
+    <h3><%= "#{@user.username}'s Games" %></h3>
```
Comment (dalexj) [inline app/views/users/show.html.erb:4]: if this is the user's page for themselves, should this say "My Games"?

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -5,4 +5,8 @@ class ApplicationController < ActionController::Base
   def current_user
     @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
   end
+
+  def require_user
+    redirect_to root_path unless current_user
```
Comment (dalexj) [inline app/controllers/application_controller.rb:10]: do you want any flash for saying "you must be logged in to be here", etc?

**Comment by dalexj** [comment]
gone through real quick, may have missed some things but I got a few. haven't looked at the javascript, its hard to really judge jquery code without pulling up the page myself in a browser

File: app/assets/javascripts/custom.js [hunk 7]
```diff
@@ -52,9 +58,19 @@ function createGame(gameParams) {
...
     success: function(newGame) {
       var listItem = "<li><a href='/" + newGame.slug + "'>" + newGame.name + "</a></li>";
       $("#one-game").append(listItem);
     },
+    error: function(xhr) {
+      var errorObject = xhr.responseJSON.errors;
+      var errorKeys = Object.keys(errorObject);
+      for(var i = 0; i < errorKeys.length; i++){
+        errorObject[errorKeys[i]].forEach(function(error) {
+          $("#welcome").append(`<span>${errorKeys[i]}:${error},  </span>`).css('background-color', '#ffcccc');
```
Comment (saylerb) [inline app/assets/javascripts/custom.js:71]: Nice use of ES6 string interpolation here. If you're mixing ES5/ES6 though, you might need to add some special stuff to spockets to make it cross-browser compatible...

**Pull request merged**